File: September Challenge/WEEK 4/C++/Day_26/Teemo_Attacking.cpp
class Solution {
public:
    int findPoisonedDuration(vector<int>& timeSeries, int duration) {
        int n   = timeSeries.size();
        int sum=0;
        if(n==0){
            return 0;
        }
        if(n==1){
            return duration;
        }
        
        for(int i=1;i<n;i++){
            int val = timeSeries[i]-timeSeries[i-1];
            sum +=min(val , duration);
        }
        sum+=duration;
        return sum;
    }
};

File: September Challenge/Week 4/Java/Day_26/TeemoAttacking.java
// Problem : https://leetcode.com/problems/teemo-attacking/
// Solution:

class Solution {
    public int findPoisonedDuration(int[] timeSeries, int duration) {
        int start = 0;
        int end=0;
        int time=0;
        for(int t : timeSeries){
            if(end<=t){
                time+=end-start;
            }else{
                time+=t-start;
            }
            start=t;
            end=t+duration;
        }
        time+=end-start;
        return time;
    }
}
